source/security/auth-mechanisms/kerberos.txt

```diff
@@ -0,0 +1,123 @@
+.. _ruby-authentication-kerberos:
+
+=================
+Kerberos (GSSAPI)
+=================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example
+
+Overview
+--------
+
+The Generic Security Services API (GSSAPI) authentication mechanism allows you to
+use your principal name to authenticate to a Kerberos service.
+You can use this mechanism only when authenticating to MongoDB Enterprise Advanced.
+
+Code Placeholders
+~~~~~~~~~~~~~~~~~
+
+The code examples on this page use the following placeholders:
+
+- ``<username>``: LDAP username.
+- ``<hostname>``: Network address of your MongoDB deployment.
+- ``<port>``: Port number of your MongoDB deployment. If you omit this parameter,
+  the driver uses the default port number (``27017``). Specifying a port number is optional
+  when connecting to a MongoDB Atlas cluster.
+- ``<authentication_db>``: MongoDB database that contains the user's LDAP credentials.
+  If you omit this parameter, the driver uses the default database (``admin``).
+
+To use the code examples on this page, replace these placeholders with your own values.
+
+Using GSSAPI Authentication in Your Application
+-----------------------------------------------
+
+To configure the MongoDB server to use Kerberos, see the
+:manual:`server Kerberos documentation
+</tutorial/control-access-to-mongodb-with-kerberos-authentication/>`.
+
+To use the Kerberos authentication mechanism with the Ruby MongoDB driver,
+you must install and load the `mongo_kerberos <https://rubygems.org/gems/mongo_kerberos>`__
+library. To do so, add the following lines to your ``Gemfile``:
+
+.. code-block:: ruby
+
+  gem 'mongo', '~> 2'
+  gem 'mongo_kerberos', '~> 2'
+
+Then, add the following lines to your application code:
+
+.. code-block:: ruby
+
+  require 'mongo'
+  require 'mongo_kerberos'
+
+If using Kerberos authentication with **MRI**, you must establish a Kerberos
+session to the driver. This session is used by the driver to prove the user's identity to
+the server. You must ensure that the host system is
+configured for Kerberos authentication. To learn more about configuring the host system
+to use Kerberos, see the `Kerberos documentation
+<https://web.mit.edu/kerberos/krb5-latest/doc/admin/install_clients.html>`__
```

Comment:
s: this seems like a second consideration, after establishing the kerberos session. maybe there could be a bulleted list of things to do: establish a session and ensure the host is configured correctly